test/table-operations.wast

```diff
@@ -0,0 +1,25 @@
+(module
```

Comment:
Can this be a lit test instead?

Reply:
I believe we haven't used lit tests for this because we depend on this type of test to do both text and binary roundtripping? But if there's a way, maybe we could convert them...

Reply:
We haven't focused on porting the existing tests of this type, but for new tests it would still be good to use lit. Using multiple run lines should be able to cover both text and binary round trips.

Reply:
I'm afraid it's not obvious to me how to do all that we need with a lit test here. Is there an existing lit test that does it?

(And, would doing this in a lit test be compatible with auto-updating it? If not, I'd rather not...)

Reply:
I think something like this should do approximately what the test runner currently does. It would also work well with the auto update script.
```
; RUN: wasm-as %s -all -g -o %t.wasm 
; RUN: wasm-dis %t.wasm -all -o %t.wast
; RUN: wasm-opt %t.wast -all -o %t.opt.wasm
; RUN: cat %t.wast | FileCheck %s
; RUN: cat %t.opt.wast | FileCheck %s --check-prefix CHECK-OPT
```

Reply:
Thanks... I still can't get it to work, though. It does read and write the file, but no `CHECK` stuff is emitted - it makes no changes at all. I spent some time now trying to figure out what's different in this file from others that do work (like having `--all-items` or not in the first line, I tried both), but didn't find anything useful. Rather than start to debug the python scripts here, I think I'll leave this for now...

Reply:
Oops, there were a few issues with that. Can you try this?

```
;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
;; RUN: wasm-as %s -all -g -o %t.wasm
;; RUN: wasm-dis %t.wasm -all -o %t.wast
;; RUN: wasm-opt %t.wast -all -o %t.opt.wasm
;; RUN: cat %t.wast | filecheck %s
;; RUN: cat %t.opt.wast | filecheck %s --check-prefix CHECK-OPT
```

Note the `;;` instead of `;` at the beginning of each line and the `filecheck` instead of `FileCheck`.

Reply:
Ah, I did fix all those issues except for the lowercase/uppercase issue with `filecheck`... now it works, thanks! Why doesn't it error though if the case is wrong?

Reply:
The update script doesn't know that `FileCheck` is not a real command, but it also doesn't recognize it as `filecheck`. I can make a PR to have it detect this error as a quality of life improvement.